### PR TITLE
chore: add dev container config

### PR DESCRIPTION
**Pull request by fgreinacher**
This is useful for people that want to to
use the VS Code Remote Development extension

### CODE REVIEW

File: .devcontainer/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+
+# To fully customize the contents of this image, use the following Dockerfile instead:
+# https://github.com/microsoft/vscode-dev-containers/tree/v0.106.0/containers/typescript-node-12/.devcontainer/Dockerfile
+FROM mcr.microsoft.com/vscode/devcontainers/typescript-node:0-12
+
+ENV DEBIAN_FRONTEND=noninteractive
+RUN apt-get update \
+   && apt-get -y install --no-install-recommends git python-minimal build-essential \
+   && apt-get autoremove -y \
+   && apt-get clean -y \
```
Comment (viceice) [inline .devcontainer/Dockerfile]:
```suggestion
```

Not needed

**Review by viceice — CHANGES_REQUESTED** [review]

File: .devcontainer/Dockerfile (hunk 1 above)
Comment (fgreinacher) [reply .devcontainer/Dockerfile]: Why not?

File: .devcontainer/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,16 @@
+#-------------------------------------------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See https://go.microsoft.com/fwlink/?linkid=2090316 for license information.
+#-------------------------------------------------------------------------------------------------------------
+
+# To fully customize the contents of this image, use the following Dockerfile instead:
+# https://github.com/microsoft/vscode-dev-containers/tree/v0.106.0/containers/typescript-node-12/.devcontainer/Dockerfile
+FROM mcr.microsoft.com/vscode/devcontainers/typescript-node:0-12
```
Comment (JamieMagee) [inline .devcontainer/Dockerfile]:
```suggestion
FROM mcr.microsoft.com/vscode/devcontainers/typescript-node:12
```

At least according to https://hub.docker.com/_/microsoft-vscode-devcontainers

File: .devcontainer/Dockerfile [hunk 3]
```diff
@@ -0,0 +1,16 @@
+#-------------------------------------------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See https://go.microsoft.com/fwlink/?linkid=2090316 for license information.
+#-------------------------------------------------------------------------------------------------------------
```
Comment (JamieMagee) [inline .devcontainer/Dockerfile]: Is this license required? As far as I can find this Dockerfile is not copied from https://github.com/microsoft/vscode-dev-containers.

File: .gitignore [hunk 4]
```diff
@@ -15,3 +15,4 @@ renovate-0.0.0-semantic-release.tgz
 junit.xml
 /test-results
 renovate-schema.json
+test/datasource/nuget/_fixtures/obj
```
Comment (JamieMagee) [inline .gitignore]: This doesn't look related to this PR.

File: .devcontainer/Dockerfile (hunk 1 above)
Comment (viceice) [reply .devcontainer/Dockerfile]: please see here: https://docs.docker.com/develop/develop-images/dockerfile_best-practices/#apt-get

File: .gitignore (hunk 4 above)
Comment (viceice) [reply .gitignore]: yes, required if you have `ms-dotnettools.csharp` installed, because it restores nuget dependencies

Comment (fgreinacher) [reply .gitignore]: Yes, it's not directly related to this PR but I considered it too tiny for a dedicated PR

File: .devcontainer/Dockerfile (hunk 3 above)
Comment (fgreinacher) [reply .devcontainer/Dockerfile]: This file was created by VS Code so I left the license.

File: .devcontainer/Dockerfile (hunk 2 above)
Comment (fgreinacher) [reply .devcontainer/Dockerfile]: This file was automatically created by VS Code. The leading zero seems to indicate the major version of the image (see https://github.com/microsoft/vscode-dev-containers/blob/master/containers/typescript-node-12/definition-build.json#L7)

File: .devcontainer/Dockerfile (hunk 1 above)
Comment (fgreinacher) [reply .devcontainer/Dockerfile]: It doesn't hurt though. But I removed it nevertheless

**Review by JamieMagee — APPROVED** [review]
